cmd/api/logging/logging.go

```diff
@@ -0,0 +1,178 @@
+	}
+	return loggingConf, nil
+}
+
+// SetLoggingCredentials sets the user and password for accessing logging backend in the request context
+func SetLoggingCredentials(loggingCreds map[string]string, loggingCredsErr error) gin.HandlerFunc {
+	user := loggingCreds[userKey]
+	password := loggingCreds[passwordKey]
+	if user == "" || password == "" || loggingCredsErr != nil {
+		loggingCredsErr = fmt.Errorf(
+			"logging secret values unset. To configure logging update the kubearchive-logging Secret")
+		return func(c *gin.Context) {
+			abort.Abort(c, loggingCredsErr, http.StatusBadRequest)
+		}
+	}
+	return func(c *gin.Context) {
+		c.Set(userKey, user)
+		c.Set(passwordKey, password)
+	}
+}
+
+// LogRetrieval retrieves a middleware function that checks the logging config and when set
+// expects to find a log url in the context from where retrieve and parse logs
+// It should be called when user, password and logURL are set in the context.
+func LogRetrieval() gin.HandlerFunc {
+	// FIXME For now the queries to the logging backend server are done insecurely. Needed for the current test env.
+	// #nosec G402
+	client := http.Client{
+		Transport: otelhttp.NewTransport(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}})}
+	return func(c *gin.Context) {
+		user := c.GetString(userKey)
+		password := c.GetString(passwordKey)
+		if user == "" || password == "" {
+			loggingCredsErr := fmt.Errorf(
+				"unexpected error. Logging credentials are unset")
+			abort.Abort(c, loggingCredsErr, http.StatusInternalServerError)
+			return
+		}
+		logUrl := c.GetString("logURL")
+		jsonPath := c.GetString("jsonPath")
+
+		if logUrl == "" {
+			abort.Abort(c, fmt.Errorf("no log URL found"), http.StatusNotFound)
+			return
+		}
+		slog.Info("Retrieving logs", "logURL", logUrl)
+
+		var jsonPathParser jp.Expr
+		var errJsonPath error
+
+		if jsonPath != "" {
+			jsonPathParser, errJsonPath = jp.ParseString(jsonPath)
+			if errJsonPath != nil {
+				abort.Abort(c, fmt.Errorf("invalid jsonPath %s for url %s", jsonPath, logUrl),
+					http.StatusInternalServerError)
+				return
+			}
+		}
+
+		request, errReq := http.NewRequest(http.MethodGet, logUrl, nil)
+		if errReq != nil {
+			abort.Abort(c, errReq, http.StatusInternalServerError)
+			return
+		}
+
+		request.SetBasicAuth(user, password)
+
+		response, errReq := client.Do(request)
+		if errReq != nil {
+			abort.Abort(c, errReq, http.StatusInternalServerError)
+			return
+		}
+
+		defer response.Body.Close()
+
+		reader := bufio.NewReader(response.Body)
+		line, readErr := readLine(reader)
+		var logsReturned bool
+		for readErr == nil {
+			parsedLine, errParseLine := parseLine(line, jsonPathParser)
+			if errParseLine != nil {
+				abort.Abort(c, errParseLine, http.StatusInternalServerError)
+				return
+			}
+			if parsedLine != "" {
+				logsReturned = true
+				c.String(http.StatusOK, parsedLine+"\n")
+			}
+			line, readErr = readLine(reader)
+		}
```

Comment:
I think this might be more clearly written like this:

```
		var logsReturned bool
		for {
			line, readErr = readLine(reader)
            if readErr != nil {
                break
            }
			parsedLine, errParseLine := parseLine(line, jsonPathParser)
			if errParseLine != nil {
				abort.Abort(c, errParseLine, http.StatusInternalServerError)
				return
			}
			if parsedLine != "" {
				logsReturned = true
				c.String(http.StatusOK, parsedLine+"\n")
			}
		}
```

Reply:
I think that your proposal is based on doing the "explicit early return" recommended by golang in general terms, but I also think that you might find the `for readErr == nil` confusing because it's a `while` hidden under the name of a `for`. It's just something I wanted to mention because I needed to do that "change of mind" when programming in go.

By the way, it's no big deal for me going with your proposal so I'll apply it.

Reply:
Change done!